neqo-bin/src/udp.rs

```diff
@@ -26,10 +26,36 @@ pub struct Socket {
 impl Socket {
     /// Create a new [`Socket`] bound to the provided address, not managed externally.
     pub fn bind<A: std::net::ToSocketAddrs>(addr: A) -> Result<Self, io::Error> {
+        const ONE_MB: usize = 1 << 20;
         let socket = std::net::UdpSocket::bind(addr)?;
+        let state = quinn_udp::UdpSocketState::new((&socket).into())?;
+
+        let send_buf_before = state.send_buffer_size((&socket).into())?;
+        // FIXME: We need to experiment if increasing this actually improves performance.
+        // Also, on BSD and Apple targets, this seems to increase the `net.inet.udp.maxdgram`
+        // sysctl, which is not the same as the socket buffer.
+        // if send_buf_before < ONE_MB {
+        //     state.set_send_buffer_size((&socket).into(), ONE_MB)?;
+        //     let send_buf_after = state.send_buffer_size((&socket).into())?;
+        //     qdebug!("Increasing socket send buffer size from {send_buf_before} to {ONE_MB}, now:
+        // {send_buf_after}"); } else {
+        //     qdebug!("Default socket send buffer size is {send_buf_before}, not changing");
+        // }
+        qdebug!("Default socket send buffer size is {send_buf_before}");
+
+        let recv_buf_before = state.recv_buffer_size((&socket).into())?;
+        if recv_buf_before < ONE_MB {
+            // Same as Firefox.
+            // <https://searchfox.org/mozilla-central/source/modules/libpref/init/StaticPrefList.yaml#13474-13478>
```

Comment:
When citing searchfox, don't use the "latest" links, use the versioned links.

https://searchfox.org/mozilla-central/rev/fa5b44a4ea5c98b6a15f39638ea4cd04dc271f3d/modules/libpref/init/StaticPrefList.yaml#13474-13477 is what you are looking for here.